pkgs/build-support/build-fhs-userenv/env.nix

```diff
@@ -124,6 +124,25 @@ let
     paths = [ etcPkg ] ++ basePkgs ++ targetPaths;
     extraOutputsToInstall = [ "out" "lib" "bin" ] ++ extraOutputsToInstall;
     ignoreCollisions = true;
+    postBuild = ''
+      if [[ -d  $out/share/gsettings-schemas/ ]]; then
+          # Create the standard schemas directory if needed
+          if [[ -L $out/share/glib-2.0 ]]; then
+              rm -rf $out/share/glib-2.0
```

Comment:
I am still not totally comfortable with removing the link. At the moment, I cannot think of a case where it would be destructive but conditions change.

Could we try to make the tree writeable like `lndir` does?

Reply:
@jtojnar could you point me to the `lndir` package you're referring to?

Reply:
It is in `xorg.lndir`.

Reply:
@jtojnar I don't see anything relevant [here](
https://github.com/NixOS/nixpkgs
/tree/b4e428ecd0535b73c42cd15ffc6028fd00a7c5ba/pkgs/servers/x11/xorg/default.nix#L1212)

Reply:
I meant the `lndir` program, not the expression.

Reply:
https://github.com/NixOS/nixpkgs/pull/161739